libkbfs/folder_branch_ops.go

```diff
@@ -4236,7 +4241,18 @@ func (fbo *folderBranchOps) SyncFromServerForTesting(
 	if err := fbo.editHistory.Wait(ctx); err != nil {
 		return err
 	}
-	return fbo.fbm.waitForQuotaReclamations(ctx)
+	if err := fbo.fbm.waitForQuotaReclamations(ctx); err != nil {
+		return err
+	}
+
+	// A second journal flush if needed, to clear out any
+	// archive/remove calls caused by the above operations.
+	if jServer, err := GetJournalServer(fbo.config); err == nil {
+		if err := jServer.Wait(context.Background(), fbo.id()); err != nil {
```

Comment:
Looks like this is a pretty common pattern now -- perhaps we need another helper function for this? Perhaps `WaitForJournal(Config, TlfID)`